lib/widgets/action_sheet.dart

```diff
@@ -402,3 +406,25 @@ class ShareButton extends MessageActionSheetMenuItemButton {
     }
   }
 }
+
+class MarkAsUnreadButton extends MessageActionSheetMenuItemButton {
+  MarkAsUnreadButton({
+    super.key,
+    required super.message,
+    required super.messageListContext,
+  }) : narrow = MessageListPage.ancestorOf(messageListContext).narrow;
```

Comment:
We want to find the narrow without calling this in the constructor or implicitly through the build method for performance reasons.

See: https://github.com/zulip/zulip-flutter/blob/dbb2dcfbf4e8e2239b2dca0142bbc3fab3cc0640/lib/widgets/message_list.dart#L48-L53